app/models/extract_fetcher.rb

```diff
@@ -65,14 +50,36 @@ def fetch_subjects?
   end
 
   def user_extracts
-    @user_extracts ||= Extract.where(filter.except(:subject_id)).order(classification_at: :desc)
+    corrected_filter = filter.except(:subject_id)
+    if fetch_minimal?
+      Extract.where(corrected_filter.merge(id: @extract_ids))
+    else
+      Extract.where(corrected_filter)
+    end
   end
 
   def subject_extracts
-    @subject_extracts ||= Extract.where(filter.except(:user_id)).order(classification_at: :desc)
+    extract_subject_ids = Extract.find(@extract_ids).pluck(:subject_id)
+    exact_subject_ids = extract_subject_ids.append(filter[:subject_id]).uniq
+    augmented_subject_ids = augment_subject_ids(exact_subject_ids)
+
+    corrected_filter = filter.except(:user_id)
+
+    if fetch_minimal?
+      # is an extract exactly in the list of extracts with the specified subject
+      # or is an extract for one of those prior subjects but not the specified subjects
+      Extract.where(corrected_filter.merge(id: @extract_ids))
+        .or(Extract.where(
+          corrected_filter.except(:subject_id).merge(subject_id: augmented_subject_ids-exact_subject_ids)
+        )
+      )
+    else
+      # is an extract for any of the subjects that were mentioned or any of their parents
+      Extract.where(corrected_filter.except(:subject_id).merge(subject_id: augmented_subject_ids))
+    end
   end
 
-  def specified_extracts
-    @specified_extracts ||= Extract.find(@extract_ids).sort_by{ |e| e.classification_at }
+  def augment_subject_ids(id_list)
+    (id_list + id_list.map{ |subject_id| Subject.find(subject_id).additional_subject_ids_for_reduction }.flatten).uniq
```

Comment:
The method name could perhaps change to `add_linked_subject_ids(subject_ids)`

Also maybe break this up to help understand what this is doing. 
```suggestion
    additional_linked_subject_ids = id_list.map do |subject_id|
      Subject.find(subject_id).additional_subject_ids_for_reduction
    end
    (id_list + additional_linked_subject_ids.flatten).uniq
```